content/docs/operating/storage.md

```diff
@@ -86,3 +86,12 @@ storage directory:
    1. Stop Prometheus.
    1. `rm -r <storage path>/*`
    1. Start Prometheus.
+
+## Hash collisions
+
+Prometheus uses 64-bit fingerprints to identify time series. On a
```

Comment:
Assuming optimal hashing, the 1e-6 probability of a collision is reached at 6.1e6 entries.
See http://en.wikipedia.org/wiki/Birthday_attack#Mathematics
Which coincides nicely with the largest usable Prometheus server.

Where is the 2^12 number for hardware or software failure from? Over which runtime?
The nice thing about the hash collision is that it doesn't increase over time if you handle the same 6.1M time series all the time.

In my previous jobs, we had data corruptions only detected by some later checksumming, quite regularly. That was a lot of data handled there, but the probability was high enough so it happened every other month or so... not one in a million...